source/Analysis/AnnualStatistics.vb

```diff
@@ -135,6 +135,38 @@ Friend Class AnnualStatistics
                 stat.avg.ToString(formatstring),
                 stat.vol.ToString(formatstring)) & eol
         Next
+
+        'Generate timeseries with max/avg/min values if years are present
+        If (Me.stats.Count > 1) Then
+            'Prepare output timeseries
+            Dim basename As String = Me.InputTimeSeries(0).Title
+            Dim timeseries_max As TimeSeries = New TimeSeries(basename + "_max")
+            Dim timeseries_avg As TimeSeries = New TimeSeries(basename + "_avg")
+            Dim timeseries_min As TimeSeries = New TimeSeries(basename + "_min")
+
+            'Set interpretation mode
+            timeseries_max.Interpretation = TimeSeries.InterpretationEnum.BlockRight
+            timeseries_avg.Interpretation = TimeSeries.InterpretationEnum.BlockRight
+            timeseries_min.Interpretation = TimeSeries.InterpretationEnum.BlockRight
+
+            'Get Unit from input timeseries
+            Dim unit As String = Me.InputTimeSeries(0).Unit
+            timeseries_max.Unit = unit
+            timeseries_avg.Unit = unit
+            timeseries_min.Unit = unit
+

```

Comment:
also set the DataSource property: `timeseries_max.DataSource = New TimeSeriesDataSource(TimeSeriesDataSource.OriginEnum.AnalysisResult)`

Reply:
Allright, I will add this!

This property is used so Wave knows where the time series comes from? What would be a common use case of this property?

Reply:
This property is displayed as Datasource in the series properties dialog. Its purpose is so that the user can see which series originated from where. When the datasource is a file, this property can also be used to reload a series from file.

Reply:
Ah I see. Thanks for this explanation!